### DISCUSSION TITLE
Link Formula from Local Installation

### DISCUSSION THREAD

**Discussion by Fray117** [Homebrew/discussions#4566] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.21-15-g05029fc
ORIGIN: https://github.com/Homebrew/brew
HEAD: 05029fc8a242101dac043fbf87fa7016bfed5818
Last commit: 3 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: edee27e6dbcce64c7e8ee2028cf992e8d6641db1
Core tap last commit: 3 days ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CURL_PATH: /usr/local/opt/curl/bin/curl
HOMEBREW_DEVELOPER: set
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.pFQbZnaxiU/org.macosforge.xquartz:0
HOMEBREW_FORCE_BREWED_CURL: set
HOMEBREW_MAKE_JOBS: 4
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_EMOJI: set
HOMEBREW_NO_INSTALL_FROM_API: set
Homebrew Ruby: 2.6.10 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: quad-core 64-bit broadwell
Clang: 12.0.0 build 1200
Git: 2.41.0 => /usr/local/bin/git
Curl: 8.1.2 => /usr/local/opt/curl/bin/curl
macOS: 10.15.7-x86_64
CLT: 12.4.0.0.1.1610135815
Xcode: N/A
```

### Output of `brew doctor`

```shell
# I'm not sure, but this specific block might related to my issue

Warning: Some installed formulae are missing dependencies.
You should `brew install` the missing dependencies:
  brew install gnu-getopt
```

### Description of issue

Hi There, I'm new on this discussion board and wonder on why I should installing a same tool twice.

I already installed `rust` using `rustup`, I happened to upgrade `gnu-getopt` which ran to installing `rust` from homebrew which I thought they would use my locally installed `rust` instead of installing a fresh formula, also it confuse me as `gnu-getopt` seems don't require `rust` but everytime I only try upgrade that formula, it automatically install `rust`, I tried to reinstall `gnu-getopt` but make me face a same problem.

**Highlighted Question:**
1. Why homebrew don't include bottled version of `rust` that already built by rust for my mac version?
2. Why I should install a new formula instead of directly link to my locally installed `rust`

**Comment by SMillerDev** [comment]
> Why homebrew don't include bottled version of rust that already built by rust for my mac version?

It is no longer supported by us or apple as would have been in `brew doctor`

> Why I should install a new formula instead of directly link to my locally installed rust

Because there is no telling if it'll work with your local install. It might be years behind the Homebrew version

**Reply by Fray117** [reply]
My local install is the latest version which same on homebrew formula previously installed using rustup directly from rust, I don't think see any significant different from the rust and homebrew.

Compatibility issue should be the user problem right? What if compatibilty check handled by the assert like homebrew did on post install (it just my opinion as it seems no harm to do so).

**Reply by SMillerDev** [reply]
> My local install is the latest version which same on homebrew formula previously installed using rustup directly from rust, I don't think see any significant different from the rust and homebrew.

But how would Homebrew be able to tell where and what version you installed of what tool? Or what architecture it is. There simply is no way for Homebrew to keep track of those kinds of things without using Homebrew packages.

> Compatibility issue should be the user problem right? What if compatibilty check handled by the assert like homebrew did on post install (it just my opinion as it seems no harm to do so).

They aren't though. If a build fails users come here for a solution and with the limited number of volunteers we have to answer questions, adding a new category of questions will only make things harder on them.

**Reply by Fray117** [reply]
> But how would Homebrew be able to tell where and what version you installed of what tool? Or what architecture it is. There simply is no way for Homebrew to keep track of those kinds of things without using Homebrew packages.

So, is it possible if I could add `rust` as homebrew formula despite locally installed?
